crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_none_literal.rs

```diff
@@ -189,40 +197,73 @@ fn create_fix_edit(
         }
     }
 
-    if literal_elements.is_empty() {
-        return Some(Edit::range_replacement(
-            "None".to_string(),
-            literal_expr.range(),
-        ));
-    }
+    let applicability = if checker.comment_ranges().intersects(literal_expr.range()) {
+        Applicability::Unsafe
+    } else {
+        Applicability::Safe
+    };
 
-    if checker.settings.target_version < PythonVersion::Py310 {
-        return None;
+    if matches!(union_kind, UnionKind::NoUnion) {
+        return Some(Fix::applicable_edit(
+            Edit::range_replacement("None".to_string(), literal_expr.range()),
+            applicability,
+        ));
     }
 
-    let bin_or = Expr::BinOp(ExprBinOp {
+    let lhs = Expr::Subscript(ast::ExprSubscript {
+        value: Box::new(literal_subscript.clone()),
         range: TextRange::default(),
-        left: Box::new(Expr::Subscript(ast::ExprSubscript {
-            value: Box::new(literal_subscript.clone()),
-            range: TextRange::default(),
-            ctx: ExprContext::Load,
-            slice: Box::new(if literal_elements.len() > 1 {
-                Expr::Tuple(ast::ExprTuple {
-                    elts: literal_elements.into_iter().cloned().collect(),
-                    range: TextRange::default(),
-                    ctx: ExprContext::Load,
-                    parenthesized: true,
-                })
-            } else {
-                literal_elements[0].clone()
-            }),
-        })),
-        op: ruff_python_ast::Operator::BitOr,
-        right: Box::new(Expr::NoneLiteral(ExprNoneLiteral {
-            range: TextRange::default(),
-        })),
+        ctx: ExprContext::Load,
+        slice: Box::new(if literal_elements.len() > 1 {
+            Expr::Tuple(ast::ExprTuple {
+                elts: literal_elements.into_iter().cloned().collect(),
+                range: TextRange::default(),
+                ctx: ExprContext::Load,
+                parenthesized: true,
+            })
+        } else {
+            literal_elements[0].clone()
+        }),
     });
+    let rhs = Expr::NoneLiteral(ExprNoneLiteral {
+        range: TextRange::default(),
+    });
+
+    match union_kind {
+        UnionKind::TypingUnion => {
+            let (import_edit, bound_name) = checker
+                .importer()
+                .get_or_import_symbol(
+                    &ImportRequest::import_from("typing", "Union"),
+                    literal_expr.start(),
+                    checker.semantic(),
+                )
+                .ok()?;
```

Comment:
nit: it would be better to return `anyhow::Result<Fix>` or `anyhow::Result<Option<Fix>>` from this function, do this here, and use `Diagnostic::try_set_fix` or `Diagnostic::try_set_optional_fix` to add the fix to the diagnostic:

```suggestion
            let (import_edit, bound_name) = checker
                .importer()
                .get_or_import_symbol(
                    &ImportRequest::import_from("typing", "Union"),
                    literal_expr.start(),
                    checker.semantic(),
                )?;
```

The advantage is that if you use `try_set_fix` or `try_set_optional_fix`, Ruff will print an error message to the terminal (if you run it with enough `-v` flags) with information from the `ResolutionError` here when this function returns an error. That makes it easier for users or us to debug why Ruff was unable to create an autofix in a specific situation.